test/jquery.i18n.test.js

```diff
@@ -392,15 +392,15 @@
 	QUnit.test( 'Support fallback loading from folder tests', function ( assert ) {
 		var i18n = $( document ).data( 'i18n' );
 
-		QUnit.stop();
+	    var done = assert.async();
 		$.when(
 			i18n.load( 'i18n/fallback', 'uk' )
 		).then( function () {
-			QUnit.start();
 			i18n.locale = 'uk';
 			assert.strictEqual( i18n.locale, 'uk', 'Locale is uk' );
 			assert.strictEqual( $.i18n( 'message_1' ), 'ONE',
 				'Message loaded from fallback locale English' );
+	        done();
```

Comment:
Legacy start/stop toggled a semaphore that was checked at the end of the test callback execution. `assert.async()` instead is meant to mark the end of the execution. Moving it lower is indeed required. However given the possibility of a failure scenario here, it should probably be passed to `.always()` not `.then()`.

In this case though, I'd recommend simply removing it alltogether and returning the promise from `$.when().then()` from the callback directly to to QUnit. Which will naturally wait for both `done` or `fail`, and also mark the test as failed if the promise is rejected.